### ISSUE TITLE
Authorization Issues - 3.7.0

### ISSUE THREAD

**Issue by DominicGunn** [postmanlabs/newman#1085]
1. Newman Version (can be found via `newman -v`): 3.7.0
 2. OS details (type, version, and architecture): Mac Sierra 10.12.5
 3. Are you using Newman as a library, or via the CLI? Library
 3. Did you encounter this recently, or has this bug always been there: Bug occurs in 3.7.0+, does not occur in 3.6.1.
 4. Expected behaviour: Enviromental variables should be injected into newman.
 5. Command / script used to run Newman: 

With Newman 3.6.X, environmental variables are injected as expected, 3.7.X environmental variables are not respected. Can be replicated using example from documentation.

```javascript
newman.run({
    ....
    environment: {
        "id": "4454509f-00c3-fd32-d56c-ac1537f31415",
        "name": "test-env",
        "values": [
            {
                "key": "foo",
                "value": "bar",
                "type": "text",
                "enabled": true
            }
        ],
        "timestamp": 1404119927461,
        "_postman_variable_scope": "environment",
        "_postman_exported_at": "2016-10-17T14:26:34.940Z",
        "_postman_exported_using": "Postman/4.8.0"
    }
})
```

**Comment by czardoz** [comment]
@DominicGunn What do you mean by "injected"? Are your variables not getting resolved? Or are they not available in the script sandbox?

**Comment by DominicGunn** [comment]
Sorry, I mean it's as if they were not being passed in through options at all, they're not available and do not resolve to anything. The behaviour is new to 3.7.X releases.

**Comment by kunagpal** [comment]
@DominicGunn I'm unable to reproduce this issue with Newman `v3.7.0` and Node v`4.x`/`6.x`. Could you also mention your Node version and a failing example? Please include the complete collection and environment file.

**Comment by DominicGunn** [comment]
Did something change with postman collections, perhaps this isn't an environmental issue? I'm using node version v7.10.0.

Our app.js runs the following code in both `3.6.1` and `3.7.0`.

```
var newman = require('newman');

newman.run({
    environment: require('./env.json'),
    collection: require('./postman_collection.json'),
    reporters: 'cli'
}, function (err) {
  console.info('Done');
});
```

Perhaps this should be made into a separate issue, here's the [postman collection](https://gist.github.com/DominicGunn/a436cb0f7bdbba37c80359c8c2438e63) and the [environmental vars](https://gist.github.com/DominicGunn/a436cb0f7bdbba37c80359c8c2438e63). 

With no changes at all in any of the code, the assertions and tests run fine in newman 3.6.1, but they fail in newman 3.7.x.

**Comment by kunagpal** [comment]
@DominicGunn It looks like the both links point to the same Gist.

**Comment by DominicGunn** [comment]
Sorry, here's the [collection](https://gist.github.com/DominicGunn/25b4d2d7616dfb94ebe9c9df232e6697)

**Comment by kunagpal** [comment]
@DominicGunn I ran the collection with the environment provided (with added logs to check the environment and request body being sent). This is what came up in the logs:
```
Newman POC

→ POST {{host}}/v1/users
  POST https://api.example.com/v1/users [errored]
  ┌
  │ 'Environment'
  │ { host: 'https://api.example.com',
  │   consumer_key: '69552',
  │   secret: 'abcd1234' }
  │ 
  │ 'Request body'
  │ '{\n  "first_name": "Newman",\n }'
  └
```

If your issue was related to the variable not being resolved in the request name itself, (`→ POST {{host}}/v1/users`), that is not currently supported.

**Comment by DominicGunn** [comment]
Is that something that is no longer supported as of 3.7.x? My confusion here stems from the fact the suite runs without failure in 3.6.1, and does fail with 3.7.x.

**Comment by kunagpal** [comment]
@DominicGunn That's weird, as sending any requests to https://api.example.com results in host resolution errors, meaning that the host probably doesn't exist anymore. I am seeing collection run failures with both Newman v3.7.0, as well as Newman v3.6.1. Could you try using https://postman-echo.com instead of https://api.example.com and see if that works for you?

**Comment by DominicGunn** [comment]
Sorry, you're right. Those collections point to non-existent endpoints. I'm not sure I could share our actual test suites.

We've been doing some digging, and it looks like under 3.7.x, requests are being rejected because of an authorization issue.

```
```- - [21/Jun/2017:15:39:38 +0000] "POST /v1/users HTTP/1.1" 401 41 "-" "PostmanRuntime/6.2.0"```
```

This issue doesn't occur under 3.6.1, and no code has changed on our service end, or in any of the tests. The only changing factor is the version of newman.

**Comment by reinouts** [comment]
We are also being affected by this issue. Authentication errors in tests suites with 3.7.0 which are running perfectly fine in 3.6.1.

**Comment by anantnaugai** [comment]
Newman 3.7.1 doesn't pass environment variables to the Authorization fields:
![image](https://user-images.githubusercontent.com/21303028/27465903-e833366c-5819-11e7-854f-d6f91448dbe0.png)
 Works perfectly fine on Postman App and Postman Runner. I had to degrade newman version to 3.5.2.

**Comment by czardoz** [comment]
This is being fixed here: https://github.com/postmanlabs/postman-runtime/pull/291

We'll release the bugfix ASAP :)

**Comment by czardoz** [comment]
Released in v3.7.2
